### ISSUE TITLE
How to enable progress bar buffering for flv from URL

### ISSUE THREAD

**Issue by shushu789** [SuRGeoNix/Flyleaf#270]
How to enable progress bar buffering for the flv from the URL (https://xxx.com/test.flv), for example, when I play the flv live broadcast of this URL address in the potplayer player, it can be played like a normal video To adjust the progress, for flv live streaming, at least you should be able to adjust the buffer position that has been played at will. I am currently using WPF. When I play the flv live stream in the URL, the progress bar slider will not move, it is always fixed at the current live time position. I'm new to Flyleaf, I checked the wiki and couldn't find instructions for that, maybe I'm missing something? Thank you！

## my development environment

* Windows 10
* visual studio 2022
* Flyleaf NuGet:FlyleafLib.Controls.WPF 1.2.10
* FFmpeg.AutoGen 6.0
* FFmpeg 6.0
![image](https://user-images.githubusercontent.com/85976805/229276766-0cb47742-adb1-4ee2-ab30-72c884d4728c.png)

**Comment by SuRGeoNix** [comment]
@shushu789 Just to confirm, when you open the input (eg. with potplayer) you can't directly seek backwards, right? Only the time that you buffer and probably after sometime it will not buffer more (eg. 2min behind?). If that's the case currently Flyleaf, does not support backwards buffering/seeking.

**Comment by shushu789** [comment]
> @shushu789确认一下，当你打开输入时（比如用potplayer）你不能直接向后搜索，对吧？只有你缓冲的时间，可能在一段时间后它不会缓冲更多（例如，落后 2 分钟？）。如果目前 Flyleaf 是这种情况，则不支持向后缓冲/查找。

Playing flv live streaming on potplayer can buffer infinitely, that is to say, as long as it keeps playing, the buffered position will always be there. We can adjust the progress slider at any time to watch the previous position. It is not 2 minutes, there is no limit. It is similar to the video in the on-demand type, which is temporarily cached locally. In this way, users can watch back and forth some previously missed clips while playing the live stream. For example, a live stream address in tiktok (https://pull-flv-f11-sg01.tiktokcdn.com/stage/stream-7216991120713075457_or4.flv) is like this.

It may be possible to implement this function by generating a temporary cache while playing. When the player window is closed or the playback is stopped, the cached data will be deleted. But I don't know how to decode these cached data while playing this live stream to achieve this goal by adjusting the position of the slider at will. I don't know what you think, and the same is true for hls streams.

* Play synchronously
![image](https://user-images.githubusercontent.com/85976805/229289348-4aa0a64e-427e-4429-bc9d-0b60beda363e.png)

* Play a minute ago
![image](https://user-images.githubusercontent.com/85976805/229289473-75d434bc-4a06-4267-ad79-5c3fd5db37f8.png)

**Comment by SuRGeoNix** [comment]
Yes, that's local caching of potplayer. Flyleaf currently does caching only for forward packets (based on Config.Demuxer.BufferDuration). I'm planning to support this in the future.

**Comment by zhouxinghong** [comment]
@SuRGeoNix Very good! 🤞😜

**Comment by zhouxinghong** [comment]
This is another player's cache mechanism, please refer to:

![Image](https://github.com/user-attachments/assets/8a181f9c-593a-4c1e-a2a2-9ba41a32ff86)

![Image](https://github.com/user-attachments/assets/ef4c8e6a-c88d-492b-adf7-e98f6f09cc8e)
